controllers/talosconfig_controller.go

```diff
@@ -38,9 +38,9 @@ import (
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/utils/pointer"
-	capiv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
```

Comment:
Nothing to major here. Use the v1alpha4 apis instead of v1alpha3 and update our handlers to the new method signatures.

Worth noting that an upgrade to this version of the bootstrap provider will/should also upgrade the CAPI core versions. This means that any existing v1alpha3 clusters, machines, etc. will get mutated to v1alpha4 on the core side. So we shouldn't have any issues using the v1alpha4 library here, but we should add testing to upgrade components and make sure it doesn't fall over.